src/js/TextCopyHelper.js

```diff
@@ -1,60 +1,99 @@
-/*global require, module*/
-"use strict";
+/**global require, module*/
+'use strict';
 
 var DomDelegate = require('ftdomdelegate');
-var ToolTip = require('./ToolTip');
-var shareDom = require('./shareDom');
+var Tooltip = require('./Tooltip');
 
+/**
+ * Gets the width of a text by using a <canvas> element
+ *
+ * @param {string} text - The text to measure
+ * @param {HTMLElement} refEl - The reference element where the text will be to get the font properties
+ *
+ * @returns {number}
+ */
+function getPixelWidthOfText(text, refEl) {
+	var c = document.createElement("canvas");
+
+	if (c.getContext && window.getComputedStyle) {
+		var ctx = c.getContext("2d");
+		var refElStyle = window.getComputedStyle(refEl);
+		ctx.font = refElStyle.getPropertyValue('font-style') + " " + refElStyle.getPropertyValue('font-size') + " " + refElStyle.getPropertyValue('font-family');
+		ctx.fillText(text, 10, 100);
+		return ctx.measureText(text).width;
+	}
+
+	return -1;
+}
+
+/**
+ * @class TextCopyHelper
+ *
+ * @param {Object} config
+ * @param {string} config.text - Value of the url input element
+ * @param {string} config.message - Tooltip text
+ * @param {HTMLElement} config.parentEl - Parent element
+ * @param {function} config.onDestroy - Optional, callback that will be ran on {@link destroy}
+ * @param {function} config.onCopy - Optional, callback that will be ran when the 'copy' event is triggered
+ */
 function TextCopyHelper(config) {
```

Comment:
The pattern I was describing comes from what I've seen @nzakas do but maybe it is an anti pattern now. With ES6 it might also become slightly outdated since parameters can be optional and have default values.